File: analyze_profile.py

import swisseph as swe
import json

# Set ephemeris path
swe.set_ephe_path('./ephe')

# Birth data: 10 June 1993, 10:57 UTC in Hamburg (lat: 53.55, lon: 10.00)
year, month, day = 1993, 6, 10
hour = 10 + 57/60
lat, lon = 53.55, 10.00

# Julian day
jd = swe.julday(year, month, day, hour)

# House system
houses, ascmc = swe.houses(jd, lat, lon, b'P')
asc = ascmc[0]

# Planet definitions
planet_ids = {
    'sun': swe.SUN,
    'moon': swe.MOON,
    'mercury': swe.MERCURY,
    'venus': swe.VENUS,
    'mars': swe.MARS,
    'jupiter': swe.JUPITER,
    'saturn': swe.SATURN,
    'uranus': swe.URANUS,
    'neptune': swe.NEPTUNE,
    'pluto': swe.PLUTO
}

# Zodiac signs
signs = [
    'Aries', 'Taurus', 'Gemini', 'Cancer', 'Leo', 'Virgo',
    'Libra', 'Scorpio', 'Sagittarius', 'Capricorn', 'Aquarius', 'Pisces'
]

# Sign to element and modality
element_map = {
    'Aries': 'fire', 'Leo': 'fire', 'Sagittarius': 'fire',
    'Taurus': 'earth', 'Virgo': 'earth', 'Capricorn': 'earth',
    'Gemini': 'air', 'Libra': 'air', 'Aquarius': 'air',
    'Cancer': 'water', 'Scorpio': 'water', 'Pisces': 'water'
}
modality_map = {
    'Aries': 'cardinal', 'Cancer': 'cardinal', 'Libra': 'cardinal', 'Capricorn': 'cardinal',
    'Taurus': 'fixed', 'Leo': 'fixed', 'Scorpio': 'fixed', 'Aquarius': 'fixed',
    'Gemini': 'mutable', 'Virgo': 'mutable', 'Sagittarius': 'mutable', 'Pisces': 'mutable'
}

# Determine sign and house
def get_sign_and_house(lon):
    sign_index = int(lon / 30)
    sign = signs[sign_index]
    for i, cusp in enumerate(houses):
        if lon < cusp:
            return sign, i if i > 0 else 12
    return sign, 12

# Collect data
profile = {
    "ascendant": signs[int(asc / 30)],
    "planets": {},
    "elements": {"fire": 0, "earth": 0, "air": 0, "water": 0},
    "modalities": {"cardinal": 0, "fixed": 0, "mutable": 0}
}

for name, pid in planet_ids.items():
    pos, _ = swe.calc_ut(jd, pid)
    lon = pos[0]
    sign, house = get_sign_and_house(lon)
    profile["planets"][name] = {"sign": sign, "house": house}
    profile["elements"][element_map[sign]] += 1
    profile["modalities"][modality_map[sign]] += 1

# Write to JSON
with open("astro_profile.json", "w") as f:
    json.dump(profile, f, indent=2)

print("✅ Profil gespeichert in astro_profile.json")

File: analyzeProfile.js
const fs = require('fs');
const path = require('path');

// Zeichen-Zuordnung für Elemente und Modalitäten
const elementsMap = {
  Aries: 'Feuer', Leo: 'Feuer', Sagittarius: 'Feuer',
  Taurus: 'Erde', Virgo: 'Erde', Capricorn: 'Erde',
  Gemini: 'Luft', Libra: 'Luft', Aquarius: 'Luft',
  Cancer: 'Wasser', Scorpio: 'Wasser', Pisces: 'Wasser'
};

const modalitiesMap = {
  Aries: 'Kardinal', Cancer: 'Kardinal', Libra: 'Kardinal', Capricorn: 'Kardinal',
  Taurus: 'Fix', Leo: 'Fix', Scorpio: 'Fix', Aquarius: 'Fix',
  Gemini: 'Beweglich', Virgo: 'Beweglich', Sagittarius: 'Beweglich', Pisces: 'Beweglich'
};

// Vereinfachte Zeichenherrscher
const rulership = {
  Sun: 'Leo', Moon: 'Cancer', Mercury: 'Gemini',
  Venus: 'Taurus', Mars: 'Aries', Jupiter: 'Sagittarius',
  Saturn: 'Capricorn', Uranus: 'Aquarius', Neptune: 'Pisces', Pluto: 'Scorpio'
};

function analyzeProfile(horoscope) {
  const elementCounts = { Feuer: 0, Erde: 0, Luft: 0, Wasser: 0 };
  const modalityCounts = { Kardinal: 0, Fix: 0, Beweglich: 0 };
  const houseCounts = Array(12).fill(0);
  const planetScores = {};

  for (const [planet, data] of Object.entries(horoscope.planets)) {
    const { sign, house } = data;

    // Element & Modalität zählen
    if (elementsMap[sign]) elementCounts[elementsMap[sign]]++;
    if (modalitiesMap[sign]) modalityCounts[modalitiesMap[sign]]++;
    if (house) houseCounts[house - 1]++;

    // Planetendominanz-Punkte
    let score = 0;
    const reasons = [];

    if (house === 1 || house === 10) {
      score += 3;
      reasons.push(`im Haus ${house}`);
    }
    if (rulership[planet] === sign) {
      score += 2;
      reasons.push(`im eigenen Zeichen (${sign})`);
    }
    const planetElement = elementsMap[sign];
    if (elementCounts[planetElement] === 1) {
      score += 1;
      reasons.push(`einziger Planet im Element ${planetElement}`);
    }

    if (score > 0) {
      planetScores[planet] = { score, reasons };
    }
  }

  // Dominanten Planet bestimmen
  let dominant = null;
  for (const [planet, { score, reasons }] of Object.entries(planetScores)) {
    if (!dominant || score > dominant.score) {
      dominant = { name: planet, score, reasons };
    }
  }

  return {
    name: horoscope.name,
    elements: elementCounts,
    modalities: modalityCounts,
    houseCluster: houseCounts
      .map((count, i) => ({ haus: i + 1, anzahl: count }))
      .filter(h => h.anzahl > 1),
    dominantPlanet: dominant || null
  };
}

// Batch-Verarbeitung für alle Nutzer:innen
const inputDir = './horoscope/';
const outputDir = './profiles/';
fs.mkdirSync(outputDir, { recursive: true });

fs.readdirSync(inputDir).forEach(file => {
  const filepath = path.join(inputDir, file);
  if (file.endsWith('.json')) {
    const data = JSON.parse(fs.readFileSync(filepath, 'utf-8'));
    const profile = analyzeProfile(data);
    const filename = `${data.name.toLowerCase().replace(/ /g, '_')}_profile.json`;
    fs.writeFileSync(path.join(outputDir, filename), JSON.stringify(profile, null, 2));
    console.log(`✅ Profil erstellt für ${data.name}`);
  }
});

